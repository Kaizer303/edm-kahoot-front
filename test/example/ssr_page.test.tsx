import SSRPage from '@/app/[locale]/ssr/page'
import { render } from '@testing-library/react'
import { useSuspenseQuery } from '@apollo/client'
import type { GetBusinessTypeResponse } from '@/app/[locale]/ssr/types'

jest.mock('@apollo/client', () => ({
  ...jest.requireActual('@apollo/client'),
  useSuspenseQuery: jest.fn(),
  gql: jest.fn(),
}))

jest.mock('next-intl', () => ({
  useLocale: jest.fn().mockReturnValue('th'),
}))

jest.mock('@/navigation', () => ({
  __esModule: true, // Required for some module systems
  Link: jest.fn(({ children, ...props }) => <div data-mock-component='Link' {...props}>{children}</div>),
}))

describe('[Page] src/app/[locale]/ssr/page.tsx', () => {
  describe('[Snapshot Test]', () => {
    it('should match no data snapshot', async () => {
      const mockData: GetBusinessTypeResponse = {
        getBusinessType: {
          data: [],
        },
      }

      const useSuspenseQueryMock = useSuspenseQuery as jest.Mock
      useSuspenseQueryMock.mockReturnValue({ data: mockData })

      const { asFragment } = render(<SSRPage />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('should match 1 data snapshot', async () => {
      const mockData: GetBusinessTypeResponse = {
        getBusinessType: {
          data: [
            {
              id: 1,
              name: 'data1',
            },
          ],
        },
      }

      const useSuspenseQueryMock = useSuspenseQuery as jest.Mock
      useSuspenseQueryMock.mockReturnValue({ data: mockData })

      const { asFragment } = render(<SSRPage />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('should match multiple data snapshot', async () => {
      const mockData: GetBusinessTypeResponse = {
        getBusinessType: {
          data: [
            {
              id: 1,
              name: 'data1',
            },
            {
              id: 2,
              name: 'data2',
            },
            {
              id: 3,
              name: 'data3',
            },
          ],
        },
      }

      const useSuspenseQueryMock = useSuspenseQuery as jest.Mock
      useSuspenseQueryMock.mockReturnValue({ data: mockData })

      const { asFragment } = render(<SSRPage />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})