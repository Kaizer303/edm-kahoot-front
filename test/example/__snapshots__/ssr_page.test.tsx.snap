// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page] src/app/[locale]/ssr/page.tsx [Snapshot Test] should match 1 data snapshot 1`] = `
<DocumentFragment>
  <h1
    class="ant-typography css-dev-only-do-not-override-mzwlov"
  >
    Server Side Rendering
  </h1>
  <div
    class="ant-divider css-dev-only-do-not-override-mzwlov ant-divider-horizontal"
    role="separator"
  />
  <div
    class="ant-row css-dev-only-do-not-override-mzwlov"
  >
    <div
      class="ant-col ant-col-xs-24 css-dev-only-do-not-override-mzwlov"
    >
      <div
        data-mock-component="Link"
        href="/ssg/1"
        locale="th"
      >
        <div>
          1 data1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[Page] src/app/[locale]/ssr/page.tsx [Snapshot Test] should match multiple data snapshot 1`] = `
<DocumentFragment>
  <h1
    class="ant-typography css-dev-only-do-not-override-mzwlov"
  >
    Server Side Rendering
  </h1>
  <div
    class="ant-divider css-dev-only-do-not-override-mzwlov ant-divider-horizontal"
    role="separator"
  />
  <div
    class="ant-row css-dev-only-do-not-override-mzwlov"
  >
    <div
      class="ant-col ant-col-xs-24 css-dev-only-do-not-override-mzwlov"
    >
      <div
        data-mock-component="Link"
        href="/ssg/1"
        locale="th"
      >
        <div>
          1 data1
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-24 css-dev-only-do-not-override-mzwlov"
    >
      <div
        data-mock-component="Link"
        href="/ssg/2"
        locale="th"
      >
        <div>
          2 data2
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-24 css-dev-only-do-not-override-mzwlov"
    >
      <div
        data-mock-component="Link"
        href="/ssg/3"
        locale="th"
      >
        <div>
          3 data3
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`[Page] src/app/[locale]/ssr/page.tsx [Snapshot Test] should match no data snapshot 1`] = `
<DocumentFragment>
  <h1
    class="ant-typography css-dev-only-do-not-override-mzwlov"
  >
    Server Side Rendering
  </h1>
  <div
    class="ant-divider css-dev-only-do-not-override-mzwlov ant-divider-horizontal"
    role="separator"
  />
  <div
    class="ant-row css-dev-only-do-not-override-mzwlov"
  />
</DocumentFragment>
`;
